src/tests/simulation/src/slt.rs

```diff
@@ -59,16 +73,54 @@ impl SqlCmd {
 }
 
 fn extract_sql_command(sql: &str) -> SqlCmd {
-    let cmd = sql
-        .trim_start()
-        .split_once(' ')
-        .unwrap_or_default()
-        .0
-        .to_lowercase();
-    match cmd.as_str() {
-        "create" => SqlCmd::Create {
-            is_create_table_as: is_create_table_as(sql),
-        },
+    let sql = sql.to_lowercase();
+    let tokens = sql.split_whitespace();
+    let mut tokens = tokens.multipeek();
+    let first_token = tokens.next().unwrap_or("");
+
+    match first_token {
+        // NOTE(kwannoel):
+        // It's entirely possible for a malformed command to be parsed as `SqlCmd::Create`.
+        // BUT an error should be expected for such a test.
+        // So we don't need to handle this case.
+        // Eventually if there are too many edge cases, we can opt to use our parser.
+        "create" => {
+            let result: Option<SqlCmd> = try {
+                match tokens.next()? {
+                    "materialized" => {
+                        // view
+                        tokens.next()?;
+
+                        // if not exists | name
+                        let next = *tokens.peek()?;
+                        if "if" == next
+                            && let Some("not") = tokens.peek().cloned()
+                            && let Some("exists") = tokens.peek().cloned() {
+                            tokens.next();
+                            tokens.next();
+                            tokens.next();
+                            let name = tokens.next()?.to_string();
+                            SqlCmd::CreateMaterializedView { name }
+                        } else {
+                            let name = next.to_string();
+                            SqlCmd::CreateMaterializedView { name }
+                        }
+                    }
+                    _ => SqlCmd::Create {
+                        is_create_table_as: is_create_table_as(&sql),
+                    },
+                }
+            };
+            result.unwrap_or(SqlCmd::Others)
+        }
+        "set" => {
+            if sql.contains("background_ddl") {
+                let enable = sql.contains("true");
+                SqlCmd::SetBackgroundDdl { enable }
+            } else {
+                SqlCmd::Others
+            }
+        }
```

Comment:
Will this be too hard-coded?

Reply:
Alternative is to use frontend to do parsing for each statement. But I choose not to do it because it may cause regression in our test runtime. So I think this is good enough for testing purpose.